src/main/scala/com/wavesplatform/state2/diffs/CommonValidation.scala

```diff
@@ -58,6 +63,25 @@ object CommonValidation {
     else Right(tx)
   }
 
+  def disallowBeforeActivationHeight[T <: Transaction](settings: FunctionalitySettings, h: Int, tx: T): Either[ValidationError, T] =
+    tx match {
+      case tx: RegisterContractTransaction if h <= settings.allowContractTransactionAfterHeight =>
```

Comment:
is tx needed here and line 70?

Reply:
since we will active the register/execute contract transaction at some height (defined in settings), so we need the check here.